service/history/api/update_workflow_util.go

```diff
@@ -88,7 +88,11 @@ UpdateHistoryLoop:
 			if err != nil {
 				return err
 			}
-			if err = NewWorkflowVersionCheck(shard, mutableState, newMutableState); err != nil {
+			lastWriteVersion, err := mutableState.GetLastWriteVersion()
```

Comment:
nit: what about update_workflow.go and signal_workflow.go

Reply:
i plan to use e.g. `signal_workflow.go` for signal API, `*_util.go` for common utility

or do you have a proposal for naming convention?

Reply:
current situation: signal & start & signal with start API share lots of common logic, but each API choose to impl the logic differently. so i plan to put all commonly used building blocks in `*_util.go`

Reply:
misunderstand your intention. I am fine with the naming